### PR TITLE
Ruby support

### LINKED ISSUE #60: Support for Ruby Projects
Can we add support for Ruby projects by analyzing Gemfiles? Ruby and Ruby on Rails project power huge portions of many companies software applications and adding support for them would be a great asset.

### PR DESCRIPTION
**Pull request by marcandre**
Fixes #60 
Builds on #72 

I didn't double check, but "peer" dependencies are dependencies of dependencies, right?

### CODE REVIEW

**Comment by znarf** [comment]
No, these are a special kind of dependencies that may not be required in the code itself, but nevertheless  needed. It's really specific to `npm` https://nodejs.org/en/blog/npm/peer-dependencies/

In `npm`, `composer` and `nuget` we're not looking at the dependencies from dependencies as they're not available in the normal dependency file (!= lock file). I haven't checked specifically at the behavior in `go`.

**Comment by marcandre** [comment]
Oh, I see.

I feel that it's a really good idea to handle indirect dependencies. There are cases of projects that may be more used by other libraries than by actual applications, so they would often not be seen by companies as stuff they use (even if indirectly) that might need funding.

Should I introduce a key for this? Maybe `sub`?

**Comment by znarf** [comment]
@marcandre The only problem I see with `sub`, it's that we're not gonna explain the dependency tree and what triggered these packages to be required.

We can try the idea starting with the `gem` implementation and try later with other managers. I'm afraid of it with `npm`, dependency trees are usually huge, we might encounter some performance issues.

Also, out of simplicity, I'm considering merging the stats for `core`, `dev` and `peer`, the differences doesn't matter really much  for our purpose.

**Comment by znarf** [comment]
@marcandre there is a minor lint issue that make test fail.

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "express": "^4.16.3",
     "express-session": "^1.15.6",
     "fs-extra": "^7.0.0",
+    "gemfile": "github:marcandre/gemfile",
```
Comment (znarf) [inline package.json]: looks like the maintainer is responsive and we will be able to get rid of that quickly

https://github.com/treycordova/gemfile/pull/2

**Comment by marcandre** [comment]
Branch rebased. Lint issue fixed. Pinged `gemfile` lib author.

**Comment by znarf** [comment]
Merging.

Will fix the linter issue on master. Will open a ticket to remember updating `gemfile` as soon as it's released.

**Pull request merged**